src/main/java/aztech/modern_industrialization/api/energy/CableTier.java

```diff
@@ -24,62 +24,65 @@
 package aztech.modern_industrialization.api.energy;
 
 import aztech.modern_industrialization.MIText;
-import net.minecraft.network.chat.Component;
+import java.util.Objects;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import net.minecraft.network.chat.MutableComponent;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
-public enum CableTier {
-    LV("LV", "lv", 32, MIText.CableTierLV),
-    MV("MV", "mv", 32 * 4, MIText.CableTierMV),
-    HV("HV", "hv", 32 * 4 * 8, MIText.CableTierHV),
-    EV("EV", "ev", 32 * 4 * 8 * 8, MIText.CableTierEV),
-    SUPERCONDUCTOR("Superconductor", "superconductor", 128000000, MIText.CableTierSuperconductor);
+/**
+ * A single tier of cables.
+ */
+public interface CableTier extends Comparable<CableTier> {
+    SortedSet<CableTier> TIERS = new TreeSet<>();
 
-    public final String englishName;
-    public final String name;
-    public final long eu;
+    /**
+     * Gets a table tier by internal name.
+     *
+     * @param name The internal name of the tier, e.g. 'LV'.
+     * @return If it exists, the added cable tier; otherwise, <code>null</code>.
+     */
+    static @Nullable CableTier getByName(@NotNull String name) {
+        for (var tier : TIERS) {
+            if (Objects.equals(tier.name(), name))
+                return tier;
+        }
 
-    public final String translationKey;
-    public final Component englishNameComponent;
+        return null;
+    }
 
-    CableTier(String englishName, String name, long eu, MIText englishNameComponent) {
-        this.englishName = englishName;
-        this.name = name;
-        this.eu = eu;
-        this.translationKey = "text.modern_industrialization.cable_tier_" + name;
-        this.englishNameComponent = englishNameComponent.text();
+    static CableTier create(String name, String englishName, long eu, MutableComponent text) {
+        return new CableTierImpl(name, englishName, eu, eu * 8, text);
     }
 
-    public static CableTier getByName(String tier) {
-        for (CableTier cableTier : values()) {
-            if (cableTier.name.equals(tier)) {
-                return cableTier;
-            }
-        }
-        return null;
+    static CableTier create(String name, String englishName, long eu, long maxTransfer, MutableComponent text) {
+        return new CableTierImpl(name, englishName, eu, maxTransfer, text);
     }
 
+    CableTier LV = create("lv", "LV", 32, MIText.CableTierLV.text());
+    CableTier MV = create("mv", "MV", 32 * 4, MIText.CableTierMV.text());
+    CableTier HV = create("hv", "HV", 32 * 4 * 8, MIText.CableTierHV.text());
+    CableTier EV = create("ev", "EV", 32 * 4 * 8 * 8, MIText.CableTierEV.text());
+    CableTier SUPERCONDUCTOR = create("superconductor", "SUPERCONDUCTOR", 128_000_000, MIText.CableTierSuperconductor.text());
```

Comment:
Something else I'll ask too is that a KubeJS event be added too in this PR, to make sure that all the functionality is exposed from the start. I think some inspiration can be drawn from the way EBF tiers are registered.